Library/Homebrew/dev-cmd/generate-formula-api.rb

```diff
@@ -69,17 +69,53 @@ def run
           File.write("_data/formula_canonical.json", "#{canonical_json}\n") unless args.dry_run?
 
           OnSystem::VALID_OS_ARCH_TAGS.each do |bottle_tag|
-            variation_formulae = all_formulae.map do |_, formula|
-              Homebrew::API.merge_variations(formula, bottle_tag:)
+            variation_formulae = all_formulae.to_h do |name, formula|
+              formula = Homebrew::API.merge_variations(formula, bottle_tag:)
+
+              version = Version.new(formula.dig("versions", "stable"))
+              pkg_version = PkgVersion.new(version, formula["revision"])
+              rebuild = formula.dig("bottle", "stable", "rebuild") || 0
+              sha256 = newest_bottle_sha256(formula, bottle_tag)
+
+              [name, [pkg_version.to_s, rebuild, sha256]]
             end
 
             unless args.dry_run?
-              File.write("api/internal/formula.#{bottle_tag}.json", JSON.pretty_generate(variation_formulae))
+              File.write("api/internal/formula.#{bottle_tag}.json", JSON.generate(variation_formulae))
             end
           end
         end
       end
 
+      sig {
+        params(formula_json: T::Hash[String, T.untyped], bottle_tag: Utils::Bottles::Tag).returns(T.nilable(String))
+      }
+      def newest_bottle_sha256(formula_json, bottle_tag)
+        available_tags = formula_json.dig("bottle", "stable", "files")&.keys&.map(&:to_sym)
+        return unless available_tags
+
+        return formula_json.dig("bottle", "stable", "files", :all, "sha256") if available_tags.include? :all
+
+        if available_tags.include? bottle_tag.to_sym
+          return formula_json.dig("bottle", "stable", "files", bottle_tag.to_sym, "sha256")
+        end
+
+        return unless bottle_tag.macos?
+
+        # If the actual tag is not available, find the newest tag with matching arch that's older than the actual tag
+        newest_viable_macos_tag = available_tags.filter_map do |tag_sym|
```

Comment:
Can you use `find_matching_tag` or `find_older_compatible_tag` for this from `Utils::Bottles` at all?

Reply:
Yep! Good call. Didn't know about these until now.